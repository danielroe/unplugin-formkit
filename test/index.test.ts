import { describe, expect, it } from 'vitest'
import unplugin from '../src/vite'
import { readFileSync } from 'fs'

type Transformer = {
  transform: (code: string, id: string) => Promise<{ code: string; map?: any }>
}
const aboutSFCFile = readFileSync('./playground/src/pages/about.vue', 'utf-8')
const contactSFCFile = readFileSync(
  './playground/src/pages/contact.vue',
  'utf-8',
)

const plugin: Transformer = unplugin() as Transformer

describe('index', () => {
  it('injects the template block into an normally structured sfc', async () => {
    expect(
      (
        await plugin.transform(
          `<template>
  <FormKit />
</template>`,
          'test.vue',
        )
      ).code,
    ).toMatchSnapshot()
  })

  it('injects inside root node if there is one', async () => {
    expect(
      (
        await plugin.transform(
          `<template>
    <div class="fizzbuzz">
      <FormKit />
    </div>
  </template>`,
          'test.vue',
        )
      ).code,
    ).toMatchSnapshot()
  })

  it('injects inside root node with full sfc', async () => {
    expect(
      (
        await plugin.transform(
          `<script lang="ts" setup>
function handleLoginSubmit(values: any) {
  window.alert("You are logged in. Credentials: \n" + JSON.stringify(values));
}
</script>

<template>
  <div>
    <FormKit type="form" submit-label="login" @submit="handleLoginSubmit">
      <FormKit type="email" label="Email" name="email" />
      <FormKit type="password" label="Password" name="password" />
    </FormKit>
  </div>
</template>
`,
          'test.vue',
        )
      ).code,
    ).toMatchSnapshot()
  })

  it('injects import into script setup block', async () => {
    expect(
      (await plugin.transform(aboutSFCFile, 'about.vue')).code,
    ).toMatchSnapshot()
  })

  it('injects setup block when using options api', async () => {
    expect(
      (await plugin.transform(contactSFCFile, 'about.vue')).code,
    ).toMatchSnapshot()
  })
})
