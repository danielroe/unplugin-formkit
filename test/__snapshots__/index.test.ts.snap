// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > injects import into script setup block 1`] = `
"<script setup lang=\\"ts\\">
import { FormKitLazyProvider } from '@formkit/vue'
import { FormKit } from '@formkit/vue'
</script>

<template>
<FormKitLazyProvider>
  <FormKit
    type=\\"text\\"
    label=\\"Your name\\"
    help=\\"Enter your name\\"
  />
</FormKitLazyProvider>
</template>"
`;

exports[`index > injects setup block when using options api 1`] = `
"<script setup lang=\\"ts\\">import { FormKitLazyProvider } from '@formkit/vue'</script>
<script lang=\\"ts\\">
import { FormKit } from '@formkit/vue'

export default {
  components: {
    FormKit
  }
}
</script>

<template>
<FormKitLazyProvider>
  <div>
    <h1>Nothing to see here</h1>
    <FormKit type=\\"text\\" label=\\"Check me out\\" />
  </div>
</FormKitLazyProvider>
</template>"
`;

exports[`index > injects the template block into an normally structured sfc 1`] = `
"<script setup>import { FormKitLazyProvider } from '@formkit/vue'</script>
<template>
<FormKitLazyProvider>
  <FormKit />
</FormKitLazyProvider>
</template>"
`;
